### PR TITLE
Use Obelisk for running reflex-examples.

### PR DESCRIPTION
**Pull request by dfordivam**
I have tried to model it in such a way that this can be integrated in reflex-frp.org as well.

Preview 
image

### CODE REVIEW

**Comment by ryantrinkle** [comment]
This looks really great!  I'll check it out ASAP.

File: frontend/src/Frontend/Examples/WebSocketChat/Main.hs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  rec
+    msgEvDyn <- widgetHold loginWidget (messagingWidget <$ loggedInEv)
+    let
+      msgSendEv = switch (current msgEvDyn)
+      msgRecEv = fmapMaybe decodeOneMsg wsRespEv
+      eRecRespTxt = fmap showMsg msgRecEv
+      loggedInEv = fmapMaybe loginEv msgRecEv
+    wsRespEv <- prerender (return never) $ do
+      let sendEv = fmap ((:[]) . toStrict . encode) msgSendEv
+      ws <- webSocket "ws://localhost:8000/websocketchat" $ def & webSocketConfig_send .~ sendEv
```
Comment (dfordivam) [inline frontend/src/Frontend/Examples/WebSocketChat/Main.hs]: Is there a way to add the web socket backend URL from Route?

**Comment by dfordivam** [comment]
PR is ready now.

I have not added 
* the othello example, it is too big and complicated.
* xhr-blob, this is covered in other examples.

File: frontend/src/Frontend/Examples/WebSocketChat/Main.hs (hunk 1 above)
Comment (luigy) [reply frontend/src/Frontend/Examples/WebSocketChat/Main.hs]: One way would be to store the host information in `config/common/route` which could later be used with `obelisk-executable-config` in the `frontend` to retrieve the files present in `config` which the `backend` serves by default. https://github.com/obsidiansystems/obelisk/tree/master/lib/executable-config#ghcjs

Did an example over here https://github.com/obsidiansystems/obelisk/commit/f87aee775be38f931fe0b5bd3c02db3aa24dfe60 which brings up that `obelisk` should get some examples as well :)

Comment (dfordivam) [reply frontend/src/Frontend/Examples/WebSocketChat/Main.hs]: Right.. `obelisk` has many stuff which would need examples.

Comment (dfordivam) [reply frontend/src/Frontend/Examples/WebSocketChat/Main.hs]: But still this is not really complete, @luigy is it possible to get the `websocketchat` part also from the `BackendRoute` ?

Comment (luigy) [reply frontend/src/Frontend/Examples/WebSocketChat/Main.hs]: @dfordivam oh I missed that it was a backend url. @cgibbard is there a convenient way to render a backend url in a similar way we can do it for the frontend?

Comment (luigy) [reply frontend/src/Frontend/Examples/WebSocketChat/Main.hs]: Covered by 2ed8d24

**Comment by luigy** [comment]
Thanks, for reporting this. For macosx the only thing I can imagine having an effect could of been https://github.com/obsidiansystems/obelisk/commit/5aeb14df78edb8dfeca62891f05d40431f4f8d53 which fixed displaying port taken errors. After bumping `obelisk` did `ob run` display that a port was already taken? I've had put a PR bumping obelisk here https://github.com/reflex-frp/reflex-examples/pull/29

**Pull request merged**